test/subdomainregistrar/TestSubdomainRegistrar.js

```diff
@@ -0,0 +1,225 @@
+    it('should allow subdomains to be created', async () => {
+      await BaseRegistrar.register(labelhash('test'), account, 86400)
+      await BaseRegistrar.setApprovalForAll(NameWrapper.address, true)
+      await NameWrapper.wrapETH2LD(
+        'test',
+        account,
+        0,
+        MAX_EXPIRY,
+        EMPTY_ADDRESS
+      )
+      expect(await NameWrapper.ownerOf(node)).to.equal(account)
+      await NameWrapper.setApprovalForAll(SubdomainRegistrar.address, true)
+      await SubdomainRegistrar2.register(
+        node,
+        'subname',
+        account2,
+        EMPTY_ADDRESS,
+        0,
+        86400,
+        []
+      )
+
+      expect(await NameWrapper.ownerOf(subNode)).to.equal(account2)
+    })
+    it('should not allow unapproved subdomains to be created', async () => {
+      await BaseRegistrar.register(labelhash('test'), account, 86400)
+      await BaseRegistrar.setApprovalForAll(NameWrapper.address, true)
+      await NameWrapper.wrapETH2LD('test', account, 0, 0, EMPTY_ADDRESS)
+      expect(await NameWrapper.ownerOf(namehash('test.eth'))).to.equal(account)
+
+      await expect(
+        SubdomainRegistrar.register(
+          namehash('test.eth'),
+          'subname',
+          account2,
+          EMPTY_ADDRESS,
+          0,
+          86400,
+          []
+        )
+      ).to.be.revertedWith(
+        `Unauthorised("${namehash('test.eth')}", "${
+          SubdomainRegistrar.address
+        }")`
+      )
+    })
+    it('should allow subdomains to be registered with a fee', async () => {
+      const node = namehash('test.eth')
+      await BaseRegistrar.register(labelhash('test'), account, 86400)
+      await BaseRegistrar.setApprovalForAll(NameWrapper.address, true)
+      await NameWrapper.wrapETH2LD('test', account, 0, 0, EMPTY_ADDRESS)
+      expect(await NameWrapper.ownerOf(node)).to.equal(account)
+      await SubdomainRegistrar.setRegistrationFee(node, 1)
+      const fee = (await SubdomainRegistrar.names(node)).registrationFee
+      await NameWrapper.setApprovalForAll(SubdomainRegistrar.address, true)
+      await SubdomainRegistrar2.register(
+        node,
+        'subname',
+        account2,
+        EMPTY_ADDRESS,
+        0,
+        86400,
+        [],
+        { value: 86400 * fee }
+      )
+
+      expect(await NameWrapper.ownerOf(subNode)).to.equal(account2)
+    })
+
+    it('should revert if insufficient ether is provided', async () => {
+      const node = namehash('test.eth')
+      await BaseRegistrar.register(labelhash('test'), account, 86400)
+      await BaseRegistrar.setApprovalForAll(NameWrapper.address, true)
+      await NameWrapper.wrapETH2LD('test', account, 0, 0, EMPTY_ADDRESS)
+      expect(await NameWrapper.ownerOf(node)).to.equal(account)
+      await SubdomainRegistrar.setupDomain(node, 1, account)
+      await NameWrapper.setApprovalForAll(SubdomainRegistrar.address, true)
+      await expect(
+        SubdomainRegistrar2.register(
```

Comment:
FYI, if this [PR](https://github.com/ensdomains/ens-contracts/pull/99) gets merged, this test will start failing as I enabled `throwOnCallFailures: false` to be able to test CCIP-read.

Please refer to [this part](https://github.com/ensdomains/ens-contracts/pull/99/files#diff-53ada1be0b3eb0cde6a37d29a3855a41bd5061aba9871e3991d1c0dbe896ee78R152) for the fix